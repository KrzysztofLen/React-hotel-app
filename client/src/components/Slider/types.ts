export type Props = {
  readonly images: Array<string>;
};
