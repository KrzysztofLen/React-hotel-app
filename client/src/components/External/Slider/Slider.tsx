import React, {Component} from "react";
import SliderItem from "./SliderItem/SliderItem";

interface IProps {
    images: Array<string> | undefined
}

interface IState {
    currentIndex: number
}

class Slider extends Component<IProps, IState> {
    constructor(props: IProps) {
        super(props);

        this.state = {
            currentIndex: 1
        }
    };

    onCloseModal(event: Event) {
        if (event != undefined) {
            event.preventDefault();
        }
    }

    onPrev = (event: Event) => {
        if (event != undefined) {
            event.preventDefault();
        }

        this.setState(prevState => ({
            currentIndex: prevState.currentIndex - 1
        }));
    }

    onNext = (event: Event) => {
        if (event != undefined) {
            event.preventDefault();
        }

        this.setState(prevState => ({
            currentIndex: prevState.currentIndex + 1
        }));
    }

    render() {
        if (this.props.images == null) {
            return false;
        }

        return (
            <SliderItem hasNext={this.state.currentIndex + 1 < this.props.images.length}
                        hasPrev={this.state.currentIndex > 0}
                        onNext={this.onNext}
                        onPrev={this.onPrev}
                        onCloseModal={this.onCloseModal}
                        src={this.props.images[this.state.currentIndex]}/>
        )
    }
}

export default Slider;