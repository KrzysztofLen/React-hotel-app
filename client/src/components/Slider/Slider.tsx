import React, { useState } from 'react';
import SliderItem from './SliderItem/SliderItem';
import { Props } from './types';

const Slider = (props: Props) => {
  const [currentIndex, setCurrentIndex] = useState<number>(1);

  const onCloseModal = (event: Event): void => {
    if (event != undefined) {
      event.preventDefault();
    }
  };

  const onPrev = (event: Event): void => {
    if (event != undefined) {
      event.preventDefault();
    }

    setCurrentIndex((prevIndex) => prevIndex - 1);
  };

  const onNext = (event: Event): void => {
    if (event != undefined) {
      event.preventDefault();
    }

    setCurrentIndex((prevIndex) => prevIndex + 1);
  };

  return (
    <SliderItem
      hasNext={currentIndex + 1 < props.images.length}
      hasPrev={currentIndex > 0}
      onNext={onNext}
      onPrev={onPrev}
      onCloseModal={onCloseModal}
      imageSrc={props.images[currentIndex]}
    />
  );
};

const MemoizedSlider = React.memo(Slider);
export { MemoizedSlider as Slider };
