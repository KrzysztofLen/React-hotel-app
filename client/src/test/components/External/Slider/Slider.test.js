import React from 'react';
import {shallow} from 'enzyme';
import Slider from '../../../../components/External/Slider/Slider';

describe("Component: Slider", () => {
	test("it should render Slider component correctly", () => {
		const wrapper = shallow(<Slider images={["1", "2"]}/>);
		expect(wrapper).toMatchSnapshot();
	});

	test("it should return false if no images provided", () => {
		const wrapper = shallow(<Slider images={null}/>);
		expect(wrapper).toMatchSnapshot();
	});

	test("it should increment state onNext", () => {
		const wrapper = shallow(<Slider images={["1", "2"]}/>);
		wrapper.setState({currentIndex: 2});
		wrapper.instance().onNext();
		wrapper.update();
		expect(wrapper.state("currentIndex")).toBe(3);
	});

	test("it should decrement state onPrev", () => {
		const wrapper = shallow(<Slider images={["1", "2"]}/>);
		wrapper.setState({currentIndex: 2});
		wrapper.instance().onPrev();
		wrapper.update();
		expect(wrapper.state("currentIndex")).toBe(1);
	});
});