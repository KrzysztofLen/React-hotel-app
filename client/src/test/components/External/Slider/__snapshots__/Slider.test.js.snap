// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Slider it should render Slider component correctly 1`] = `
<SliderItem
  hasNext={false}
  hasPrev={true}
  onCloseModal={[Function]}
  onNext={[Function]}
  onPrev={[Function]}
  src="2"
/>
`;

exports[`Component: Slider it should return false if no images provided 1`] = `""`;
